src/shared/extensionUtilities.ts

```diff
@@ -343,14 +366,52 @@ export function getToolkitEnvironmentDetails(): string {
     const vsCodeVersion = vscode.version
     const envDetails = localize(
         'AWS.message.toolkitInfo',
-        'OS:  {0} {1} {2}\n{3} Extension Host Version:  {4}\nAWS Toolkit Version:  {5}\n',
+        'OS:  {0} {1} {2}\n{3} Extension Host Version:  {4}\{5} Toolkit Version:  {6}\n',
         osType,
         osArch,
         osRelease,
         getIdeProperties().longName,
         vsCodeVersion,
+        getIdeProperties().company,
         pluginVersion
     )
 
     return envDetails
 }
+
+/**
+ * Returns the Cloud9 compute region or 'unknown' if we can't pull a region, or `undefined` if this is not Cloud9.
+ */
+ export async function initializeComputeRegion(
+    metadataService: MetadataService = new MetadataService(),
+    isC9: boolean = isCloud9()
+): Promise<void> {
+    if (isC9) {
+        try {
+            const identity: { [key: string]: string; region: string } = await new Promise((resolve, reject) => {
+                // MetadataService.request() doesn't support .promise()...
+                metadataService.request('/latest/dynamic/instance-identity/document', (err, data) => {
+                    if (err) {
+                        reject(err)
+                    }
+                    resolve(JSON.parse(data))
+                })
+            })
+
+            computeRegion = identity.region || 'unknown'
+        } catch (e) {
+            computeRegion = 'unknown'
+        }
+    } else {
+        // not cloud9, region has no meaning
+        computeRegion = undefined
+    }
+}
+
+export function getComputeRegion(): string | undefined {
+    if (computeRegion === NOT_INITIALIZED) {
+        throw new Error('Attempted to get compute region without initializing.')
```

Comment:
I'm thinking this segment unfortunately is a little too cute for its own good (minver tests are failing and it can cause loading issues elsewhere). Thinking that's an endorsement for https://github.com/aws/aws-toolkit-vscode/pull/1786/files#r644379477 , but it does bring up some questions:
* do we care if our metrics around compute region aren't backwards compatible after this change?
* this might cause issues with some calls that set constants depending on the region, in that if they get initialized prior to the compute region being determined, they'll remain that way for the lifetime of the extension. Is this an acceptable risk?
* There is also a risk of this popping in at a later point: some text might read as "AWS" followed by "Amazon" later.

Reply:
This is necessary although I'll have to find a way around it on minver tests. The extension was breaking due to strings being localized and stored as consts prior to extension activation.

Reply:
We probably should force "Amazon" if the compute region is not initialized. So it'd be Amazon -> AWS instead of the inverse in the worst case.

Reply:
We can't do that when we're initializing as constants since we won't be able to change the value back.

Reply:
Hmm that's unfortunate. The 'proper' way to do it then would be to have an event for compute region initialization that modules register with. When fired, they do whatever is needed to update the relevant strings.

I think the above event-driven solution is OOS and adds unnecessary complexity. The current proposal works fine as long as no constant is set forever (i.e. it will eventually update after the compute region is discovered)